wsproto/connection.py

```diff
@@ -275,6 +275,11 @@ def events(self):
                 if frame.opcode is Opcode.PING:
                     assert frame.frame_finished and frame.message_finished
                     self._outgoing += self._proto.pong(frame.payload)
+                    yield PingReceived(frame.payload)
+
+                elif frame.opcode is Opcode.PONG:
+                    assert frame.frame_finished and frame.message_finished
+                    yield PongReceived(frame.payload)
```

Comment:
No, we definitely can't verify the payload. RFC 6455 says:

> A Pong frame MAY be sent unsolicited.  This serves as a unidirectional heartbeat.

So if we get a Pong frame with an unexpected payload, we have no way to know whether that's because the peer screwed up the payload, or because they just happened to spontaneously send us a weird Pong and the one we're expecting will be along any moment. So even if we stored the payload, we couldn't do anything with that information.